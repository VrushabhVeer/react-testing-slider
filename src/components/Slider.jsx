import React, { useEffect, useState } from "react";

const Slider = () => {
  const [page, setPage] = useState(1);
  const [questions, setQuestions] = useState([]);

  useEffect(() => {
    fetch(`http://localhost:8080/questions?page=${page}_&_limit=${1}`)
      .then((res) => res.json())
      .then((res) => {
        setQuestions(res);
      });
  }, [page]);

  return (
    <div data-testid="sliderComponent">
      {questions.map((q) => (
        <div key={q.id}>
          <p>Que : {q.question}</p>
          <p>Ans : {q.answer}</p>
        </div>
      ))}

      <div>
        <button disabled={page === 0} onClick={() => setPage(page - 1)}>
          Prev
        </button>
        <button disabled={page === 3} onClick={() => setPage(page + 1)}>
          Next
        </button>
      </div>
    </div>
  );
};

export default Slider;
