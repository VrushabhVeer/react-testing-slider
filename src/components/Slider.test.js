import Slider from "./Slider";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

describe("Testing the Slider Component", () => {
  it("Should be present in the DOM", () => {
    render(<Slider />);

    let slider = screen.getByTestId("sliderComponent");
    expect(slider).toBeInTheDocument();
  });

  it("Should be present in the DOM", () => {
    render(<Slider />);
    let prevButton = screen.getByText("Prev");

    expect(prevButton).toBeInTheDocument();
  });

  it("Should be present in the DOM", () => {
    render(<Slider />);
    let nextButton = screen.getByText("Next");

    expect(nextButton).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Slider />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render children", () => {
    render(<App />);

    let slider = screen.getByTestId("sliderComponent");
    expect(slider).toBeInTheDocument();
  });
});
