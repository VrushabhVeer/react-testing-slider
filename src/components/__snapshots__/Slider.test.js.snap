// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Slider Component should match the snapshot 1`] = `
<div
  data-testid="sliderComponent"
>
  <div>
    <button
      disabled={false}
      onClick={[Function]}
    >
      Prev
    </button>
    <button
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
